org.eclipse.xtext/src/org/eclipse/xtext/documentation/impl/MultiLineJavaDocTypeReferenceProvider.java

```diff
@@ -44,6 +63,48 @@
 		return regions;
 	}
 
+	/**
+	 * Returns the referenced element in javadoc comment at a given <i>offset</i>.
+	 * 
+	 * @since 2.16
+	 */
+	@Override
+	public EObjectInComment getEObjectReferencedInComment(XtextResource resource, int offset) {
+		IParseResult parseResult = resource.getParseResult();
+		if(parseResult != null) {
+			INode rootNode = parseResult.getRootNode();
+			INode node = NodeModelUtils.findLeafNodeAtOffset(rootNode, offset);
+			EObject semanticObject = NodeModelUtils.findActualSemanticObjectFor(node);
+			if(semanticObject != null) {
+				EReference reference = getEReference();
+				if(reference!=null) {
+					IScope scope = scopeProvider.getScope(semanticObject, reference);
+					List<ReplaceRegion> typeReferences = computeTypeRefRegions(node);
+					for(ReplaceRegion typeReference : typeReferences) {
+						if(typeReference.getOffset() <= offset && offset <= typeReference.getEndOffset()) {
+							String typeReferenceText = typeReference.getText();
+							if(typeReferenceText != null && typeReferenceText.length() > 0) {
+								ITextRegion region = new TextRegion(typeReference.getOffset(), typeReference.getLength());
+								IEObjectDescription candidate = scope.getSingleElement(qualifiedNameConverter.toQualifiedName(typeReferenceText));
```

Comment:
Please extract a method to query the scope. Implementations should have access to the entire comment node, and the region